api_definition_test.go

```diff
@@ -437,3 +435,71 @@ func TestRoundRobin(t *testing.T) {
 		}
 	}
 }
+
+func setupKeepalive(conn net.Conn) error {
+	tcpConn := conn.(*net.TCPConn)
+	if err := tcpConn.SetKeepAlive(true); err != nil {
+		return err
+	}
+	if err := tcpConn.SetKeepAlivePeriod(30 * time.Second); err != nil {
+		return err
+	}
+	return nil
+}
+
+type customListener struct {
+	L net.Listener
```

Comment:
I just realized that I can't just embedd the field, because I need to override Accept function. Also `ListenAddr` is a go-rpc function, and this is actually is not net.Listener but gorpc.Listener interface, which differs.